src/models/client.ts

```diff
@@ -11,3 +11,73 @@ export interface GetInvoicesArgs {
   limit?: number;
   offset?: number;
 }
+
+interface CreateInvoiceArgsBase {
+  currency: string;
+  price: string | number;
+  orderId?: string | number;
+  expirationTime?: string;
+  itemDesc?: string;
+  itemCode?: string;
+  posData?: string;
+  status?: string;
+  redirectUrl?: string;
+  transactionSpeed?: string;
+  physical?: boolean;
+  minerFees?: {
+    [index: string]: {
+      satoshisPerByte: string | number;
+      totalFee: string | number;
+    };
+  };
+  supportedTransactionCurrencies?: {
+    [index: string]: {
+      enabled: boolean;
+    };
+  };
+  exchangeRates?: {
+    [index: string]: {
+      [index: string]: string | number;
+    };
+  };
+  refundable?: boolean;
+  taxIncluded?: string | number;
+  nonce?: string | number;
+  guid?: string;
+  token?: string;
+  redirectAutomatically?: boolean;
+  notificationEmail?: string;
+  notificationURL?: string;
+  extendedNotifications?: boolean;
+  fullNotifications?: boolean;
+}
+
+interface CreateInvoiceArgsBuyerObj extends CreateInvoiceArgsBase {
+  buyer?: {
+    name: string;
+    address1?: string;
+    address2?: string;
+    locality?: string;
+    region?: string;
+    postalCode?: string;
+    country?: string;
+    phone?: string;
+    email?: string;
+  };
+}
+
+interface CreateInvoiceArgsBuyerSep extends CreateInvoiceArgsBase {
+  buyerEmail?: string;
+  buyerPhone?: string;
+  buyerCountry?: string;
+  buyerZip?: string;
+  buyerState?: string;
+  buyerCity?: string;
+  buyerAddress2?: string;
+  buyerAddress1?: string;
+  buyerName?: string;
+}
+
+export type CreateInvoiceArgs =
```

Comment:
So the request use `CreateInvoiceArgsBuyerSep` but the response is using different structure similar to `CreateInvoiceArgsBuyerObj `. Because this PR is only about the Request, you should just put the fields of `CreateInvoiceArgsBuyerSep ` directly inside the request type.